dotnet/samples/HomeAutomation/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ Copyright (c) Microsoft. All rights reserved.
+
+ Example that demonstrates how to use Semantic Kernel in conjunction with dependency injection.
+
+ Loads app configuration from:
+ - appsettings.json.
+ - appsettings.{Environment}.json.
+ - Secret Manager when the app runs in the "Development" environment (set through the DOTNET_ENVIRONMENT variable).
+ - Environment variables.
+ - Command-line arguments.
+*/
+
+using HomeAutomation.Options;
+using HomeAutomation.Plugins;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Http.Resilience;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace HomeAutomation;
+
+internal static class Program
+{
+    internal static async Task Main(string[] args)
+    {
+        HostApplicationBuilder builder = Host.CreateApplicationBuilder(args);
+
+        // Actual code to execute is found in Worker class
+        builder.Services.AddHostedService<Worker>();
+
+        // Get configuration
+        builder.Services.AddOptions<AzureOpenAI>()
+                        .Bind(builder.Configuration.GetSection(nameof(AzureOpenAI)))
+                        .ValidateDataAnnotations()
+                        .ValidateOnStart();
+
```

Comment:
Could we also show adding a logging service?

Reply:
That's the magic of DI: it's already added automagically!